diff --git org.eclipse.epsilon.eugenia.examples.executablestatemachine.graphical.diagram/src/esm/diagram/part/DeleteElementAction.java b/examples/org.eclipse.epsilon.eugenia.examples.executablestatemachine.graphical.diagram/src/esm/diagram/part/DeleteElementAction.java
index 8bb754f..d13e864 100644
--- org.eclipse.epsilon.eugenia.examples.executablestatemachine.graphical.diagram/src/esm/diagram/part/DeleteElementAction.java
+++ org.eclipse.epsilon.eugenia.examples.executablestatemachine.graphical.diagram/src/esm/diagram/part/DeleteElementAction.java
@@ -27,10 +27,11 @@
 public class DeleteElementAction extends AbstractDeleteFromAction {
 
 	/**
-	 * @generated
+	 * @generated NOT
 	 */
 	public DeleteElementAction(IWorkbenchPart part) {
 		super(part);
+		System.out.println("Testing patch");
 	}
 
 	/**